src/coreclr/jit/gentree.cpp

```diff
@@ -12495,28 +12495,44 @@ GenTree* Compiler::gtFoldExpr(GenTree* tree)
                 {
                     fgWalkTreePre(&colon_op2, gtClearColonCond);
                 }
-
+#ifdef DEBUG
+                if (verbose)
+                {
+                    printf("\nIdentical GT_COLON trees! ");
+                    gtDispTree(op2);
+                }
+#endif
+                GenTree* op;
                 if (sideEffList == nullptr)
                 {
                     // No side-effects, just return colon_op2
-                    return colon_op2;
+                    JITDUMP("No side effects, bashing to second operand:\n")
+                    op = colon_op2;
                 }
                 else
                 {
 #ifdef DEBUG
                     if (verbose)
                     {
-                        printf("\nIdentical GT_COLON trees with side effects! Extracting side effects...\n");
+                        printf("Extracting side effects...\n");
                         gtDispTree(sideEffList);
-                        printf("\n");
                     }
 #endif
                     // Change the GT_COLON into a GT_COMMA node with the side-effects
                     op2->ChangeOper(GT_COMMA);
                     op2->gtFlags |= (sideEffList->gtFlags & GTF_ALL_EFFECT);
                     op2->AsOp()->gtOp1 = sideEffList;
-                    return op2;
+
+                    JITDUMP("Transformed GT_COLON into GT_COMMA:\n");
+                    op = op2;
                 }
+#ifdef DEBUG
+                if (verbose)
+                {
+                    gtDispTree(op);
+                }
+#endif
+                return op;
```

Comment:
It is notable that this part of the change is most involved and at the same time probably least useful. Quick survey of places where `QMARK` nodes are created suggests they are quite unlikely to end up with identical trees for the `COLON`.

Ominously, commenting out this optimization results in `0` diffs across the framework assemblies when crossgening...